ui/packages/catalog-ui-search/src/main/webapp/react-component/location/bounding-box.js

```diff
@@ -143,21 +146,91 @@ const BoundingBoxUtm = (props) => {
     );
 };
 
+const BoundingBoxDms = (props) => {
+    const {
+        dmsSouth,
+        dmsNorth,
+        dmsWest,
+        dmsEast,
+
+        dmsSouthDirection,
+        dmsNorthDirection,
+        dmsWestDirection,
+        dmsEastDirection,
+
+        cursor
+    } = props;
+
+    const latitudeDirections = ['N', 'S'];
+    const longitudeDirections = ['E', 'W'];
+
+    return (
+        <div className="input-location">
+            <DmsLongitude
```

Comment:
❗️ All of the `<DmsLongtitude>` sections are rendering the same thing save for binding different labels, values, etc. I recommend creating a list of associated arrays and mapping that into the components:

```
const options = [{label: 'West', value: dmsWest, direction:...}, {label: 'South', ...}, {label: 'East', ...}];

return <div className="input-location">
 options.map(item => 
 <DmsLongtitude label={item.label} value={item.value} onChange={cursor(item.cursorValue)}>
  <Direction options={item.directionOption} value={item.directionValue}.../>
 </DmsLongtitude>
);
</div>
```

Reply:
Two of the inputs are `DmsLatitude` and two are `DmsLongitude`. Is it possible to specify the component names dynamically like this?

Reply:
Looks like it is possible: https://reactjs.org/docs/jsx-in-depth.html#choosing-the-type-at-runtime

I will try this out.

Reply:
[This](https://gist.github.com/jrnorth/1937a8d01c173acc7a4c074ddf9e332d) is a working version I came up with. It's both longer (if attempting to maintain a similar level of readability) and more complex than laying out the components statically. I don't see the benefit of doing it this way but I'd like to hear some other opinions. And maybe my gist can be simplified.